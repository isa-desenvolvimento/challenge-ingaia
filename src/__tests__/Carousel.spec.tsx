import { render } from '@testing-library/react';
import Carousel from '../components/Carousel';

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {
    images: {
      domains: ['cdn.example.com'],
    },
  },
}));

jest.mock('next/image', () => ({ src, alt }) => <img src={src} alt={alt} />);

describe('<Carousel />', () => {
  const images = [
    { url: 'http://http2.mlstatic.com/D_844171-MLA41757309022_052020-O.jpg', id: '844171-MLA41757309022_052020' },
  ];
  it('should render component Carousel children', () => {
    const { container } = render(
      <Carousel images={images} />,
    );
    expect(container).toBeTruthy();
  });

  it('should render snapshot', () => {
    const { container } = render(
      <Carousel images={images} />,
    );
    expect(container).toMatchSnapshot();
  });
});
