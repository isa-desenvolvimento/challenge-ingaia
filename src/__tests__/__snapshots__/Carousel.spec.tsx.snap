// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> should render snapshot 1`] = `
<div>
  <div
    class="carousel-root"
    tabindex="0"
  >
    <div
      class="carousel carousel-slider"
      style="width: 100%;"
    >
      <ul
        class="control-dots"
      >
        <li
          aria-label="slide item 1"
          class="dot selected"
          role="button"
          tabindex="0"
          value="0"
        />
      </ul>
      <button
        aria-label="previous slide / item"
        class="control-arrow control-prev control-disabled"
        type="button"
      />
      <div
        class="slider-wrapper axis-horizontal"
        style="height: auto;"
      >
        <ul
          class="slider animated"
          style="transform: translate3d(0,0,0); transition-duration: 350ms; height: auto;"
        >
          <li
            class="slide selected"
          >
            <div
              class="container"
            >
              <img
                alt="844171-MLA41757309022_052020"
                src="http://http2.mlstatic.com/D_844171-MLA41757309022_052020-O.jpg"
              />
            </div>
          </li>
        </ul>
      </div>
      <button
        aria-label="next slide / item"
        class="control-arrow control-next control-disabled"
        type="button"
      />
    </div>
  </div>
</div>
`;
